### PR TITLE
Create test coverage reports

### PR DESCRIPTION
**Pull request by luishfonseca**

### CODE REVIEW

File: core/tests/CMakeLists.txt [hunk 1]
```diff
@@ -1,8 +1,6 @@
 # core/tests/CMakeLists.txt
 # Core tests build configuration
 
-include(CTest)
-
```
Comment (RiscadoA) [inline core/tests/CMakeLists.txt:5]: Whats the reasoning behind this?

File: CMakeLists.txt [hunk 2]
```diff
@@ -9,6 +9,23 @@ set(CMAKE_MODULE_PATH ${CMAKE_MODULE_PATH} "${CMAKE_SOURCE_DIR}/cmake/")
...
+    setup_target_for_coverage_lcov(
+        NAME coverage-core
+        EXECUTABLE core/tests/cubos-core-tests
+        DEPENDENCIES cubos-core-tests
+
+        EXCLUDE "core/lib/*" "core/tests/*"
+
+        LCOV_ARGS "--no-external"
+    )
+endif()
```
Comment (RiscadoA) [inline CMakeLists.txt]: Shouldn't this be moved to the `core` CMakeLists.txt?

Comment (luishfonseca) [reply CMakeLists.txt]: in the future I also want to expand this to the engine, whenever we start unit testing it as well.
it has already been renamed to "coverage" and the executable could be changed to `core/tests/cubos-core-tests && engine/tests/cubos-engine-tests`

File: core/tests/CMakeLists.txt (hunk 1 above)
Comment (luishfonseca) [reply core/tests/CMakeLists.txt:5]: idk, it didn't look necessary and everything still work after removing it. if I shouldn't have removed it lmk why

Comment (RiscadoA) [reply core/tests/CMakeLists.txt:5]: I thought it was necessary for add_tests /shrug

**Review by RiscadoA — APPROVED** [review]

**Review by RiscadoA — APPROVED** [review]

**Pull request merged**